src/emulator/storage/files.ts

```diff
@@ -221,8 +221,12 @@ export class StorageLayer {
     if (!upload) {
       return undefined;
     }
+
+    if (upload.status !== UploadStatus.FINISHED) {
+      this._persistence.deleteFile(upload.fileLocation);
+    }
     upload.status = UploadStatus.CANCELLED;
```

Comment:
Only ACTIVE uploads should be cancellable. I believe the spec for cancel to be:

1. if active -> 200
2. if cancelled -> 200
3. if finished -> 400
4. if non-existent -> 404

Reply:
Done.